components/range_cache_memory_engine/src/background.rs

```diff
@@ -1141,11 +1151,31 @@ impl Runnable for DeleteRangeRunner {
     fn run(&mut self, task: Self::Task) {
         match task {
             BackgroundTask::DeleteRange(ranges) => {
+                fail::fail_point!("on_in_memory_engine_delete_range");
                 let (mut ranges_to_delay, ranges_to_delete) = {
                     let core = self.engine.read();
                     let mut ranges_to_delay = vec![];
                     let mut ranges_to_delete = vec![];
                     for r in ranges {
+                        // Check whether range exists in `ranges_being_deleted` and it's scheduled
+                        if !core.range_manager.ranges_being_deleted.iter().any(
+                            |(range_being_delete, scheduled)| {
+                                if range_being_delete == &r {
+                                    if !scheduled {
+                                        panic!(
+                                            "range to delete with scheduled false; range={:?}",
+                                            r,
+                                        );
+                                    }
+                                    true
+                                } else {
+                                    false
+                                }
```

Comment:
```suggestion
                                if range_being_delete == &r && !scheduled {
                                    panic!(
                                        "range to delete with scheduled false; range={:?}",
                                        r,
                                    );
                                };
                                range_being_delete == &r
```

Let's reduce indents.